ipn/ipnlocal/profiles.go

```diff
@@ -278,19 +312,32 @@ func newUnusedID(knownProfiles map[ipn.ProfileID]*ipn.LoginProfile) (ipn.Profile
 	}
 }
 
-// setPrefsLocked sets the current profile's prefs to the provided value.
-// It also saves the prefs to the StateStore, if the current profile
-// is not new.
-func (pm *profileManager) setPrefsLocked(clonedPrefs ipn.PrefsView) error {
-	pm.prefs = clonedPrefs
-	pm.updateHealth()
-	if pm.currentProfile.ID == "" {
-		return nil
+// setProfilePrefs sets the profile's prefs to the provided value.
+// If the profile is not new, it saves the prefs to the [ipn.StateStore].
+// It returns an error if the profile is non-current and does not have its Key
+// set, or if the prefs could not be saved.
+func (pm *profileManager) setProfilePrefsNoCheck(profile *ipn.LoginProfile, clonedPrefs ipn.PrefsView) error {
```

Comment:
also what does `NoCheck` mean

Reply:
I updated the docstring and renamed it `setProfilePrefsUnchecked`.